From 2c696255749683e9a084f797eb033d222510a275 Mon Sep 17 00:00:00 2001
From: Akira TAGOH <akira@tagoh.org>
Date: Mon, 18 Feb 2013 13:17:53 +0900
Subject: [PATCH 1/2] Bug 60748 - broken conf.d/10-autohint.conf and
 conf.d/10-unhinted.conf

Move the target of recipes to the "pattern" from the "font".
This is to ensure the targeted objects is updated by them
prior to FcDefaultSubstitute() so that it can adds the default
values properly.
---
 conf.d/10-autohint.conf          | 2 +-
 conf.d/10-no-sub-pixel.conf      | 2 +-
 conf.d/10-sub-pixel-bgr.conf     | 2 +-
 conf.d/10-sub-pixel-rgb.conf     | 2 +-
 conf.d/10-sub-pixel-vbgr.conf    | 2 +-
 conf.d/10-sub-pixel-vrgb.conf    | 2 +-
 conf.d/10-unhinted.conf          | 2 +-
 conf.d/11-lcdfilter-default.conf | 2 +-
 conf.d/11-lcdfilter-legacy.conf  | 2 +-
 conf.d/11-lcdfilter-light.conf   | 2 +-
 10 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/conf.d/10-autohint.conf b/conf.d/10-autohint.conf
index bf08de8..f9032ba 100644
--- a/conf.d/10-autohint.conf
+++ b/conf.d/10-autohint.conf
@@ -2,7 +2,7 @@
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
 <!--  Use the Autohinter --> 
-  <match target="font">
+  <match target="pattern">
     <!--
       This configuration is available on the major desktop environments.
       We shouldn't overwrite it with "assign" unconditionally.
diff --git a/conf.d/10-no-sub-pixel.conf b/conf.d/10-no-sub-pixel.conf
index e650160..cf12465 100644
--- a/conf.d/10-no-sub-pixel.conf
+++ b/conf.d/10-no-sub-pixel.conf
@@ -2,7 +2,7 @@
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
 <!--  Enable sub-pixel rendering --> 
-  <match target="font">
+  <match target="pattern">
     <!--
       This configuration is available on the major desktop environments.
       We shouldn't overwrite it with "assign" unconditionally.
diff --git a/conf.d/10-sub-pixel-bgr.conf b/conf.d/10-sub-pixel-bgr.conf
index 74574ad..1378195 100644
--- a/conf.d/10-sub-pixel-bgr.conf
+++ b/conf.d/10-sub-pixel-bgr.conf
@@ -2,7 +2,7 @@
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
 <!--  Enable sub-pixel rendering --> 
-  <match target="font">
+  <match target="pattern">
     <!--
       This configuration is available on the major desktop environments.
       We shouldn't overwrite it with "assign" unconditionally.
diff --git a/conf.d/10-sub-pixel-rgb.conf b/conf.d/10-sub-pixel-rgb.conf
index 3c558e5..757c3b6 100644
--- a/conf.d/10-sub-pixel-rgb.conf
+++ b/conf.d/10-sub-pixel-rgb.conf
@@ -2,7 +2,7 @@
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
 <!--  Enable sub-pixel rendering --> 
-  <match target="font">
+  <match target="pattern">
     <!--
       This configuration is available on the major desktop environments.
       We shouldn't overwrite it with "assign" unconditionally.
diff --git a/conf.d/10-sub-pixel-vbgr.conf b/conf.d/10-sub-pixel-vbgr.conf
index 392634d..c807e5c 100644
--- a/conf.d/10-sub-pixel-vbgr.conf
+++ b/conf.d/10-sub-pixel-vbgr.conf
@@ -2,7 +2,7 @@
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
 <!--  Enable sub-pixel rendering --> 
-  <match target="font">
+  <match target="pattern">
     <!--
       This configuration is available on the major desktop environments.
       We shouldn't overwrite it with "assign" unconditionally.
diff --git a/conf.d/10-sub-pixel-vrgb.conf b/conf.d/10-sub-pixel-vrgb.conf
index 1b7b5f6..b5985e0 100644
--- a/conf.d/10-sub-pixel-vrgb.conf
+++ b/conf.d/10-sub-pixel-vrgb.conf
@@ -2,7 +2,7 @@
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
 <!--  Enable sub-pixel rendering --> 
-  <match target="font">
+  <match target="pattern">
     <!--
       This configuration is available on the major desktop environments.
       We shouldn't overwrite it with "assign" unconditionally.
diff --git a/conf.d/10-unhinted.conf b/conf.d/10-unhinted.conf
index 67037b4..a25e3b1 100644
--- a/conf.d/10-unhinted.conf
+++ b/conf.d/10-unhinted.conf
@@ -2,7 +2,7 @@
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
 <!--  Disable hinting --> 
-  <match target="font">
+  <match target="pattern">
     <!--
       This configuration is available on the major desktop environments.
       We shouldn't overwrite it with "assign" unconditionally.
diff --git a/conf.d/11-lcdfilter-default.conf b/conf.d/11-lcdfilter-default.conf
index e251edd..9d7d11f 100644
--- a/conf.d/11-lcdfilter-default.conf
+++ b/conf.d/11-lcdfilter-default.conf
@@ -2,7 +2,7 @@
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
 <!--  Use lcddefault as default for LCD filter -->
-  <match target="font">
+  <match target="pattern">
     <!--
       This configuration is available on the major desktop environments.
       We shouldn't overwrite it with "assign" unconditionally.
diff --git a/conf.d/11-lcdfilter-legacy.conf b/conf.d/11-lcdfilter-legacy.conf
index 63a4116..7e2b256 100644
--- a/conf.d/11-lcdfilter-legacy.conf
+++ b/conf.d/11-lcdfilter-legacy.conf
@@ -2,7 +2,7 @@
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
 <!--  Use lcdlegacy as default for LCD filter -->
-  <match target="font">
+  <match target="pattern">
     <!--
       This configuration is available on the major desktop environments.
       We shouldn't overwrite it with "assign" unconditionally.
diff --git a/conf.d/11-lcdfilter-light.conf b/conf.d/11-lcdfilter-light.conf
index af50b2c..0dd0a90 100644
--- a/conf.d/11-lcdfilter-light.conf
+++ b/conf.d/11-lcdfilter-light.conf
@@ -2,7 +2,7 @@
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
 <!--  Use lcdlight as default for LCD filter -->
-  <match target="font">
+  <match target="pattern">
     <!--
       This configuration is available on the major desktop environments.
       We shouldn't overwrite it with "assign" unconditionally.
-- 
1.8.1.5

